### PR TITLE
SWC-3137: auto-hide project tabs where there is no content.

### PR DESCRIPTION
**Pull request by jay-hodgson**
Show feature selection on initial project creation

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/display/ProjectDisplayDialog.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+			updateCache(view.getChallenge(), result.challengeHasContent(), CHALLENGE);
+			updateCache(view.getDiscussion(), result.discussionHasContent(), DISCUSSION);
+			updateCache(view.getDocker(), result.dockerHasContent(), DOCKER);
+		} catch (IllegalArgumentException e) {
+			synAlert.showError(e.getMessage());
+			return;
+		}
+		hide();
+	}
+
```
Comment (kimyen) [inline src/main/java/org/sagebionetworks/web/client/widget/display/ProjectDisplayDialog.java:122]: What happen when a user create a project, chose Wiki and File, but has not added any content, then leave. Later, he/she log on to Synapse via a different machine?

File: src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java [hunk 2]
```diff
@@ -3145,4 +3155,131 @@ public ColumnModelPage getPossibleColumnModelsForViewScope(ViewScope scope, Stri
...
+	
+	@Override
+	public boolean isWiki(String projectId) throws RestServiceException {
+		org.sagebionetworks.client.SynapseClient synapseClient = createSynapseClient();
+		return isWiki(projectId, synapseClient);
+	}
+	
+	private boolean isForum(String projectId, org.sagebionetworks.client.SynapseClient synapseClient) throws SynapseException {
+		Forum forum = synapseClient.getForumByProjectId(projectId);
+		if (forum == null) {
```
Comment (kimyen) [inline src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java:3233]: the backend will always create a forum when user asks for it on a valid projectId.

File: src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java [hunk 3]
```diff
@@ -3145,4 +3155,131 @@ public ColumnModelPage getPossibleColumnModelsForViewScope(ViewScope scope, Stri
...
+			return false;
+		}
+		return synapseClient.getThreadCountForForum(forum.getId(), DiscussionFilter.NO_FILTER).getCount() > 0;
```
Comment (kimyen) [inline src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java:3236]: DiscussionFilter.NO_FILTER > what if there was a thread and a thread has been deleted? Assuming that the user delete the thread to be able to hide a tab? 

NO_FILTER allows retrieving both deleted and non-deleted thread. 
Suggest to use EXCLUDE_DELETED

File: src/main/java/org/sagebionetworks/web/client/widget/display/ProjectDisplayDialog.java (hunk 1 above)
Comment (jay-hodgson) [reply src/main/java/org/sagebionetworks/web/client/widget/display/ProjectDisplayDialog.java:122]: They will be presented with the project display settings again (on the other machine)

File: src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java (hunk 2 above)
Comment (jay-hodgson) [reply src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java:3233]: will fix...

File: src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java (hunk 3 above)
Comment (jay-hodgson) [reply src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java:3236]: will fix...

**Pull request merged**